### PR TITLE
Purge pre-🍋 navigation 

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Following #7203 and #7059 all users on Cloud/the latest version are using the navigation redesign. This means we can remove the old code and rename `lemonade` to just `navigation` as planned.

## How did you test this code?

Relying on existing automated tests, but also clicked around manually. Made sure that `WhoAmI` is fine with being moved around by trying out the invite flow too.

### CODE REVIEW

**Review by paolodamico — APPROVED** [review]
lgtm! non-blocking comments inline

File: frontend/src/layout/navigation/navigationLogic.ts [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+import api from '../../lib/api'
+import { FEATURE_FLAGS } from '../../lib/constants'
+import { featureFlagLogic } from '../../lib/logic/featureFlagLogic'
+import { eventUsageLogic } from '../../lib/utils/eventUsageLogic'
+import { systemStatusLogic } from '../../scenes/instance/SystemStatus/systemStatusLogic'
+import { organizationLogic } from '../../scenes/organizationLogic'
+import { preflightLogic } from '../../scenes/PreflightCheck/logic'
+import { teamLogic } from '../../scenes/teamLogic'
+import { userLogic } from '../../scenes/userLogic'
+import { VersionType } from '../../types'
```
Comment (mariusandra) [inline frontend/src/layout/navigation/navigationLogic.ts:12]: @Twixes It's a nit, but I often see your IDE work against some other IDEs or configs in a `short<->relative` import renaming battle royale.

```diff
+import { teamLogic } from '../../scenes/teamLogic'
-import { teamLogic } from 'scenes/teamLogic'
```

Personally I'd use the shorter ones, if there's a defined alias (in tsconfig.json) in the path like `scenes/` or `lib/` or `~/`.

File: frontend/src/layout/navigation/navigationLogic.ts [hunk 2]
```diff
@@ -20,62 +22,107 @@ type WarningType =
...
+        hideProjectSwitcher: true,
         setHotkeyNavigationEngaged: (hotkeyNavigationEngaged: boolean) => ({ hotkeyNavigationEngaged }),
-        setProjectModalShown: (isShown: boolean) => ({ isShown }),
-        setOrganizationModalShown: (isShown: boolean) => ({ isShown }),
     },
     reducers: {
-        menuCollapsed: [
-            typeof window !== 'undefined' && window.innerWidth <= 991,
+        isSideBarShownRaw: [
+            window.innerWidth >= 992, // Sync width threshold with Sass variable $lg!
```
Comment (mariusandra) [inline frontend/src/layout/navigation/navigationLogic.ts:49]: Any sense to put this in [`windowValues`](https://keajs.org/docs/plugins/window-values/)?

**Review by mariusandra — APPROVED** [review]
Looks good. Two random comments inside.

File: frontend/src/layout/navigation/navigationLogic.ts [hunk 3]
```diff
@@ -1,14 +1,16 @@
+import dayjs from 'dayjs'
```
[the remark this answers is not in the document]
Comment (Twixes) [reply frontend/src/layout/navigation/navigationLogic.ts:1]: Yeah, now that I think about it it'd probably be better for the diff to do the file rename in a separate PR, but it's just `lemonadeLogic` and `navigationLogic` being fully merged with most stuff coming from the newer one.

File: frontend/src/layout/navigation/navigationLogic.ts (hunk 2 above)
Comment (Twixes) [reply frontend/src/layout/navigation/navigationLogic.ts:49]: Oh, didn't know about `windowValues`! Though it seems this doesn't work for a reducer's default state. I'll definitely use it when resolving #7091

File: frontend/src/layout/navigation/navigationLogic.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/layout/navigation/navigationLogic.ts:12]: I agree, it's just my VS Code set up doing not taking `tsconfig` into account 😅 I'll look into making it use the shorter paths

Comment (Twixes) [reply frontend/src/layout/navigation/navigationLogic.ts]: Turns out for some reason I had this set to the non-default "relative" value:
:O
You can now sleep well knowing that my future PRs will have the shortest imports possible. 😄

**Pull request merged**